### DISCUSSION TITLE
ControlNet from the Extensions menu

### DISCUSSION THREAD

**Discussion by Nacho-sh** [zero01101/openOutpaint#250] [Q&A] [answered]
So I have learned to use this tool recently and I have to say it's pretty awesome. However, there is something that I haven't been able to work out: how the heck does the ControlNet "magic and promptless inpainting and outpainting" work here? In the readme it is claimed that you can use it to basically expand the images to infinity without a prompt, but all I get when I turn the "Controlnet In/Outpainting" checkbox on, is a random image following a box of noise. 
Am I supposed to do anything apart from setting the parameters in OpenOutpaint? I would really appreciate any guide on how to outpaint using ControlNet in this extension ^^
![Untitled](https://github.com/zero01101/openOutpaint/assets/75551026/02ff0003-db59-49a4-bf9d-229a74103afe)

**Comment by zero01101** [comment, marked as answer]
hey, thanks :) so yeah, the 3 links under the "magical promptless inpainting and outpainting" bit ([a](https://github.com/Mikubill/sd-webui-controlnet/discussions/1464), [b](https://github.com/Mikubill/sd-webui-controlnet/discussions/1143), [c](https://github.com/Mikubill/sd-webui-controlnet/discussions/1597)) go to the relevant discussions in the controlnet extension repo which has probably more information than necessary but goes over the relevant options pretty well.  sorry the docs haven't been updated in a while, kinda just scrambled those updates out 

tl;dr though, i'd say start with preprocessor inpaint_only+lama, mode +cn (aka controlnet is more important), and resize mode fill, and uncheck reference as 1st unit; that generally only works "well" for pure img2img/inpainting and isn't even a good idea all the time then either.  not quite sure what i was thinking adding that one...  inpaint global harmonious preprocessor is particularly good for pure inpainting tasks too

you'll also probably have worse-than-optimal luck with a 384x resolution, it definitely works better on a 512x area at least :/ 

anyway, video examples using no prompts and a non-inpainting checkpoint
outpainting: 

https://github.com/zero01101/openOutpaint/assets/1649724/4fbbc27f-9e34-46a8-8f6d-b0718587c9cd

inpainting:

https://github.com/zero01101/openOutpaint/assets/1649724/b90a8e08-38bd-48aa-8f08-7313708781aa

**Comment by Nacho-sh** [comment]
Thanks a lot for your reply. It was just my dumbass brain which forgot to download the proper ControlNet yaml file to make it work; definitely not the extension's fault. Sorry for wasting your time, but hopefully this will encourage people to use this extension, seeing the amazing results a promptless try can give. Keep up the good work!
